openedx/core/djangoapps/courseware_api/tests/test_views.py

```diff
@@ -156,3 +156,54 @@ def test_resume_with_completion(self):
         assert response.data['block_id'] == str(self.unit.location)
         assert response.data['unit_id'] == str(self.unit.location)
         assert response.data['section_id'] == str(self.sequence.location)
+
+
+@ddt.ddt
+class CelebrationApiTestViews(BaseCoursewareTests):
+    """
+    Tests for the resume API
+    """
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.url = '/api/courseware/celebration/{}'.format(cls.course.id)
+
+    def setUp(self):
+        super().setUp()
+        self.enrollment = CourseEnrollment.enroll(self.user, self.course.id, 'verified')
+
+    @ddt.data(True, False)
+    def test_happy_path(self, update):
+        if update:
+            CourseEnrollmentCelebrationFactory(enrollment=self.enrollment, celebrate_first_section=False)
+
+        response = self.client.post(self.url, {'first_section': True}, content_type='application/json')
+        assert response.status_code == 200
+
+        celebration = CourseEnrollmentCelebration.objects.first()
+        assert celebration.celebrate_first_section
+        assert celebration.enrollment.id == self.enrollment.id
+
+    def test_extra_data(self):
+        response = self.client.post(self.url, {'extra': True}, content_type='application/json')
+        assert response.status_code == 400
+
+    def test_no_data(self):
+        response = self.client.post(self.url, {}, content_type='application/json')
+        assert response.status_code == 200
+        assert CourseEnrollmentCelebration.objects.count() == 0
+
+    def test_no_enrollment(self):
+        self.enrollment.delete()
+        response = self.client.post(self.url, {'first_section': True}, content_type='application/json')
+        assert response.status_code == 404
+
+    def test_no_login(self):
+        self.client.logout()
+        response = self.client.post(self.url, {'first_section': True}, content_type='application/json')
+        assert response.status_code == 401
+
+    def test_no_course(self):
+        response = self.client.post('/api/courseware/celebration/course-v1:foo+bar+1T2020',
```

Comment:
Super extra nit: change from `foo+bar` to `fake+course` to be even more explicit about this being fake and not just a test course (since foo and bar show up a lot in tests)

Reply:
Done.